mne/preprocessing/ica.py

```diff
@@ -1486,21 +1488,33 @@ def _pick_sources(self, data, include, exclude):
         logger.info('Zeroing out %i ICA components'
                     % (n_components - len(sel_keep)))
 
-        unmixing = np.eye(_n_pca_comp)
-        unmixing[:n_components, :n_components] = self.unmixing_matrix_
-        unmixing = np.dot(unmixing, self.pca_components_[:_n_pca_comp])
+        if (self.method == 'imported_eeglab' and
+           np.diff(self.unmixing_matrix_.shape)[0]):
+            # When \PCA reduction is used in EEGLAB, runica returns
+            # weights= weights*sphere*eigenvectors(:,1:ncomps)';
+            # sphere = eye(urchans);
+            # such that we cannot separate the unmixing matrix and
+            # the PCA components. For that reason, the composite
+            # unmixing matrix takes a rectangular shape.
+            unmixing = self.unmixing_matrix_
```

Comment:
I am surprised there is no dc offset involved.

Reply:
As far as I know, EEGLAB does not export the dc offset. Anyway, the dc offset is just something that should be removed so that there is no offset. Maybe I am missing a point about this DC offset, but it does not make much sense to apply a DC offset for an ICA computed on some data to another dataset which is most likely to have another offset. The offset just need to be canceled out by .apply(), no?

Reply:
ok then EEGLAB does not store the `pca_mean_` and recomputes or not the centering on the new data on which it's applied.